lib/index.js

```diff
@@ -66,6 +68,8 @@ var subtag = require("subtag");
 
 function MaplibreGeocoder(geocoderApi, options) {
   this._eventEmitter = new EventEmitter();
+  options.showResultsWhileTyping =
+    !!geocoderApi.getSuggestions || options.showResultsWhileTyping; // Force showResultsWhileTyping to be on when suggestions API is passed in
```

Comment:
This is confusing for users to provide an explicit flag and then override it internally. Can we at least mark the flag deprecated and add this internal logic in its documentation if we don't want to make breaking changes.

Reply:
The field won't be deprecated for usage of this library without the suggestions API, we still need to support the flag for devs using the geocode APIs only. Should we then just remove this logic and default `showResultsWhileTyping` to `true` is a `getSuggestions` API is passed in?

Reply:
Thadd and I talked about this a bit and we decided that one path here would be to just leave `showResultsWhileTyping` explicitly for geocode results and to add a flag that deals with just suggestion results. The outcome of that work is this incoming PR: https://github.com/maplibre/maplibre-gl-geocoder/pull/29

Reply:
This code block was removed for this release.